### PR TITLE
Add Ballerina By Example Guide on Code to Cloud

### PR DESCRIPTION
**Pull request by xlight05**
## Purpose
$Subject

### CODE REVIEW

File: c2c-extension-examples/examples/c2c-deployment/c2c_deployment.bal [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import ballerina/http;
+import ballerina/log;
+import ballerina/c2c as _;
+
+//Note that code is completely focused on business logic and it does not specify anything related to operations.
```
Comment (praneesha) [inline c2c-extension-examples/examples/c2c-deployment/c2c_deployment.bal:5]:
```suggestion
// Note that the code below is completely focused on the business logic and it does not specify anything related to operations.
```
Need to add a space after the '//' sign. Need to fix in all occurrences.

File: c2c-extension-examples/examples/c2c-deployment/c2c_deployment.description [hunk 2]
```diff
@@ -0,0 +1,5 @@
+//Ballerina supports generating docker and Kubernetes artifacts from code without any additional configuration.
```
Comment (praneesha) [inline c2c-extension-examples/examples/c2c-deployment/c2c_deployment.description:1]:
```suggestion
// Ballerina supports generating Docker and Kubernetes artifacts from code without any additional configuration.
```

File: c2c-extension-examples/examples/c2c-deployment/c2c_deployment.description [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+//This simplifies the developer experience of developing and deploying ballerina code in the cloud
```
Comment (praneesha) [inline c2c-extension-examples/examples/c2c-deployment/c2c_deployment.description:2]:
```suggestion
//This simplifies the experience of developing and deploying Ballerina code in the cloud.
```

File: c2c-extension-examples/examples/c2c-deployment/c2c_deployment.description [hunk 4]
```diff
@@ -0,0 +1,5 @@
...
+//Code to cloud builds containers and required artifacts by deriving required values from the code
```
Comment (praneesha) [inline c2c-extension-examples/examples/c2c-deployment/c2c_deployment.description:3]:
```suggestion
//Code to cloud builds the containers and required artifacts by deriving the required values from the code.
```

File: c2c-extension-examples/examples/c2c-deployment/c2c_deployment.description [hunk 5]
```diff
@@ -0,0 +1,5 @@
...
+//If you want to override the default values taken from the compiler, you can use Ballerina.cloud file. <br/><br/>
```
Comment (praneesha) [inline c2c-extension-examples/examples/c2c-deployment/c2c_deployment.description:4]:
```suggestion
//If you want to override the default values taken by the compiler, you can use a `Ballerina.cloud` file. <br/><br/>
```

File: c2c-extension-examples/examples/c2c-deployment/c2c_deployment.description [hunk 6]
```diff
@@ -0,0 +1,5 @@
...
+//For more information, see the [Code to Cloud Deployment Guide](https://ballerina.io/swan-lake/learn/deployment/code-to-cloud).
```
Comment (praneesha) [inline c2c-extension-examples/examples/c2c-deployment/c2c_deployment.description:5]:
```suggestion
//For more information, see [Code to Cloud Deployment](/swan-lake/learn/deployment/code-to-cloud).
```

File: c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out [hunk 7]
```diff
@@ -0,0 +1,82 @@
+# In order to override the default values given by the compiler, we’ll be making Ballerina.cloud file in the project directory. This follows toml rules. We’ll use the following values for simplicity. All the supported key value properties can be found in code to cloud specification.
```
Comment (praneesha) [inline c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out:1]:
```suggestion
# In order to override the default values taken by the compiler, you need to have the `Ballerina.cloud` file in the project directory. This follows the general TOML rules. The following values are used for simplicity. All the supported key value properties can be found in the code to cloud specification.
```

File: c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+
+[container.image]
+repository="wso2inc"
+name="hello"
+tag="v0.1.0"
+
+# Note that ballerina build should be executed in the project directory. Code to Cloud generates only one container per project.
```
Comment (praneesha) [inline c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out:8]:
```suggestion
# Note that the `ballerina build` command should be executed from the project directory. Code to cloud generates only one container per project.
```

File: c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+        target/bin/hello.jar
+
+Generating artifacts...
+
+        @kubernetes:Service                      - complete 1/1
+        @kubernetes:Deployment                   - complete 1/1
+        @kubernetes:HPA                          - complete 1/1
+        @kubernetes:Docker                       - complete 2/2
+
+        Execute the below command to deploy the Kubernetes artifacts:
```
Comment (praneesha) [inline c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out:32]:
```suggestion
        Execute the command below to deploy the Kubernetes artifacts:
```

File: c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+        kubectl apply -f /home/wso2/test/c2c-slp6/target/kubernetes/hello
+
+        Execute the below command to access service via NodePort:
```
Comment (praneesha) [inline c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out:35]:
```suggestion
        Execute the command below to access the service via NodePort:
```

File: c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+        kubectl expose deployment wso2-hello-0--deployment --type=NodePort --name=wso2-hello-0--svc-local
+
+
+
+# Let’s try executing the docker image separately first.
```
Comment (praneesha) [inline c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out:40]:
```suggestion
# Let’s try executing the Docker image separately first.
```

File: c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+
+# Verify if the Docker image is generated
```
Comment (praneesha) [inline c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out:42]:
```suggestion
# Verify if the Docker image is generated.
```

File: c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+$ docker images
+
+REPOSITORY                    TAG                 IMAGE ID            CREATED             SIZE
+wso2inc/hello                       v0.1.0              60d95f0928b2        About a minute ago   228MB
+
+# Run the generated docker image
```
Comment (praneesha) [inline c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out:48]:
```suggestion
# Run the generated Docker image.
```

File: c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+$ docker run -d -p 9090:9090 wso2inc/hello:v0.1.0
+c04194eb0b4d0d78cbc8ca55e0527d381d8ab4a1a68f8ea5dd3770a0845d5fbb
+
+# Access the Service
```
Comment (praneesha) [inline c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out:52]:
```suggestion
# Access the service.
```

File: c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+$ curl http://localhost:9090/helloWorld/sayHello
+Hello, World from service helloWorld !
```
Comment (praneesha) [inline c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out:54]:
```suggestion
Hello, World from the `helloWorld` service!
```

File: c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+
+# Lets try executing Kubernetes service now.
```
Comment (praneesha) [inline c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out:56]:
```suggestion
# Lets try executing the Kubernetes service now.
```

File: c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+$ kubectl apply -f /home/wso2/test/c2c-slp6/target/kubernetes/hello
+service/helloep-svc created
+deployment.apps/wso2-hello-0--deployment created
+horizontalpodautoscaler.autoscaling/wso2-hello-0--hpa created
+
+# Verify kubernetes pods
```
Comment (praneesha) [inline c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out:62]:
```suggestion
# Verify the Kubernetes pods.
```

File: c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+$ kubectl get pods
+NAME                                          READY   STATUS    RESTARTS   AGE
+wso2-hello-0--deployment-7d4d56457b-7jlzx   1/1     Running   0          57s
+
+# Expose via nodeport to test in the developer environment.
```
Comment (praneesha) [inline c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out:67]:
```suggestion
# Expose via NodePort to test in the developer environment.
```

File: c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out [hunk 19]
```diff
@@ -0,0 +1,82 @@
...
+$ kubectl expose deployment wso2-hello-0--deployment --type=NodePort --name=wso2-hello-0--svc-local
+service/wso2-hello-0--svc-local exposed
+
+#Get ip and port of the kubernetes service.
```
Comment (praneesha) [inline c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out:71]:
```suggestion
#Get the IP and port of the Kubernetes service.
```

File: c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out [hunk 20]
```diff
@@ -0,0 +1,82 @@
...
+$ kubectl get svc
+NAME                        TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)          AGE
+wso2-hello-0--svc-local   NodePort    10.111.61.112    <none>        9090:32437/TCP   4m17s
+
+# If you are using minikube, IP address should be changed according to the output of the `minikube ip` command.
```
Comment (praneesha) [inline c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out:76]:
```suggestion
# If you are using Minikube, the IP address should be changed according to the output of the `minikube ip` command.
```

File: c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out [hunk 21]
```diff
@@ -0,0 +1,82 @@
...
+$ minikube ip
+192.168.49.2
+
+# Access the deployed service via curl
```
Comment (praneesha) [inline c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out:80]:
```suggestion
# Access the deployed service via cURL.
```

File: c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out [hunk 22]
```diff
@@ -0,0 +1,82 @@
...
+$ curl http://192.168.49.2:32437/helloWorld/sayHello
+Hello, World from service helloWorld !
```
Comment (praneesha) [inline c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out:82]:
```suggestion
Hello, World from the `helloWorld` service!
```

File: c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out (hunk 9 above)
Comment (xlight05) [reply c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out:32]: This is the output from ballerina compilation. I think we should fix it there before fixing it here

File: c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out (hunk 10 above)
Comment (xlight05) [reply c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out:35]: This is the output from ballerina compilation. I think we should fix it there before fixing it here

File: c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out (hunk 15 above)
Comment (xlight05) [reply c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out:54]: This is the string from the response.setTextPayload("Hello, World from service helloWorld ! ");. 
in c2c_deployment.bal. 
I dont think we should highlight helloWorld. What about others? should i change?

File: c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out (hunk 22 above)
Comment (xlight05) [reply c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out:82]: This is the string from the response.setTextPayload("Hello, World from service helloWorld ! ");.
in c2c_deployment.bal.
I dont think we should highlight helloWorld. What about others? should i change?

Comment (praneesha) [reply c2c-extension-examples/examples/c2c-deployment/c2c_deployment.out:84]: That's fine, then.

**Review by praneesha — APPROVED** [review]

**Pull request merged**